com.unity.ml-agents.extensions/Runtime/Match3/Move.cs

```diff
@@ -201,6 +205,21 @@ public static Move FromPositionAndDirection(int row, int col, Direction dir, int
             };
         }
 
+        /// <summary>
+        /// Check if the move is valid for the given board size.
+        /// This will be passed the return value from AbstractBoard.GetCurrentBoardSize().
+        /// </summary>
+        /// <param name="boardSize"></param>
+        /// <returns></returns>
+        public bool IsValidForBoardSize(BoardSize boardSize)
```

Comment:
I feel like since a move is considered valid or invalid only in the context of a board, this check should go in the board.

Reply:
yeah I agree with this, sounds more natural to have Board.IsValidMove(Move move)

Reply:
Talked to @surfnerd offline, we agreed on `InRangeForBoard`. I wanted to separate moved that we know are "invalid" because they go off the current board size, vs. moves that are invalid due to specific game logic.